### PR TITLE
refactor: Merge _remove_agent into remove_agent

### PR DESCRIPTION
**Pull request by rht**
Depends on #1241 to be merged.

### CODE REVIEW

File: mesa/space.py [hunk 1]
```diff
@@ -479,9 +474,9 @@ def move_to_empty(
                     break
         else:
             new_pos = agent.random.choice(sorted(self.empties))
+        self.remove_agent(agent)
```
Comment (rht) [inline mesa/space.py:477]: Caution when reviewing this. I intentionally move this line to be before the `_place_agent`.

**Comment by rht** [comment]
CI is failing because this needs to be rebased on top of #1241.

**Comment by Corvince** [comment]
I haven't reviewed, but I think this is a very good idea.

**Comment by rht** [comment]
Rebased.

**Review by tpike3 — APPROVED** [review]
LGTM

**Pull request merged**